src/main/java/com/wynntils/modules/core/commands/CommandServer.java

```diff
@@ -65,6 +75,82 @@ public void execute(MinecraftServer server, ICommandSender sender, String[] args
         }
     }
 
+    private void nextSoulPoint(MinecraftServer server, ICommandSender sender, String[] args){
+        Map<String, Integer> nextServers = new HashMap<>();
+
+        for (String availableServer : ServerListManager.getAvailableServers().keySet()) {
+            ServerProfile serverProfile = ServerListManager.getServer(availableServer);
+            int uptimeMinutes = serverProfile.getUptimeMinutes();
+            nextServers.put(availableServer, 20 - (uptimeMinutes % 20));
+        }
+
+        // Sort servers by time until soul point
+        Map<String, Integer> sortedServers = nextServers.entrySet().stream()
+                .sorted(Map.Entry.comparingByValue())
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (e1, e2) -> e1, LinkedHashMap::new));
+
+        if (args[1].equalsIgnoreCase("switch") || args[1].equalsIgnoreCase("s")) {
+            if (WebManager.getPlayerProfile().canUseSwitch()) {
+                String world = sortedServers.keySet().stream().limit(1).toString();
+                try {
+                    sender.getServer().getCommandManager().executeCommand(sender, "switch " + world);
+                } catch (NullPointerException e) {
+                    server.createCommandManager().executeCommand(sender, "switch " + world);
+                }
+            }
+            else {
```

Comment:
My bad on this, should have explained this better.

```suggestion
} else {
```